pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -59,13 +62,35 @@ func (r *checker) Check(ctx context.Context) error {
 		return err
 	}
 
-	if ingress.Spec.DefaultCertificate == nil {
-		return errNoDefaultCertificate
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	return validateCertificate(cv.Spec.ClusterID, ingress.Spec.DefaultCertificate, cluster.Spec.Domain)
+}
+
+func validateCertificate(clusterId configv1.ClusterID, ingressCertificate *corev1.LocalObjectReference, clusterDomain string) error {
+	isManagedDomain := dns.IsManagedDomain(clusterDomain)
+
+	if ingressCertificate == nil {
+		if isManagedDomain {
+			return errNoCertificateAndManagedDomain
+		}
+		return errNoCertificateAndCustomDomain
 	}
 
-	if ingress.Spec.DefaultCertificate.Name != string(cv.Spec.ClusterID)+"-ingress" {
-		return fmt.Errorf("custom ingress certificate in use: %q", ingress.Spec.DefaultCertificate.Name)
+	certificateName := ingressCertificate.Name
+	isCertificateNameValid := isCertificateNameValid(certificateName, clusterId)
```

Comment:
```suggestion
```
I think you could leave this variable out, it's clear from the name of the function what it's doing here, and it's only used once in the code below.

Reply:
I agree on this one. It will simplify the logic a bit.

Reply:
Done. Tks!